Tests break with Go 1.5.1

```
TESTING...
=== RUN   TestGearmanSource
panic: runtime error: index out of range

goroutine 7 [running]:
gopkg.in/Clever/optimus.v3/sources/gearman.TestGearmanSource.func2(0xc82000a740)
        /Users/joshdeare/src/go/src/gopkg.in/Clever/optimus.v3/sources/gearman/gearman_test.go:57 +0xc67
created by gopkg.in/Clever/optimus.v3/sources/gearman.TestGearmanSource
        /Users/joshdeare/src/go/src/gopkg.in/Clever/optimus.v3/sources/gearman/gearman_test.go:61 +0x7d0

goroutine 1 [chan receive]:
testing.RunTests(0x476ec8, 0x59d2a0, 0x2, 0x2, 0x1)
        /usr/local/Cellar/go/1.5.1/libexec/src/testing/testing.go:562 +0x8ad
testing.(*M).Run(0xc82005def8, 0x0)
        /usr/local/Cellar/go/1.5.1/libexec/src/testing/testing.go:494 +0x70
main.main()
        gopkg.in/Clever/optimus.v3/sources/gearman/_test/_testmain.go:56 +0x116

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
        /usr/local/Cellar/go/1.5.1/libexec/src/runtime/asm_amd64.s:1696 +0x1

goroutine 5 [chan receive]:
gopkg.in/Clever/optimus.v3/tests.GetRows(0x6b78e8, 0xc82000a7c0, 0x0, 0x0, 0x0)
        /Users/joshdeare/src/go/src/gopkg.in/Clever/optimus.v3/tests/test_table.go:32 +0xb5
gopkg.in/Clever/optimus.v3/sources/gearman.TestGearmanSource(0xc82007e120)
        /Users/joshdeare/src/go/src/gopkg.in/Clever/optimus.v3/sources/gearman/gearman_test.go:62 +0x82e
testing.tRunner(0xc82007e120, 0x59d2a0)
        /usr/local/Cellar/go/1.5.1/libexec/src/testing/testing.go:456 +0x98
created by testing.RunTests
        /usr/local/Cellar/go/1.5.1/libexec/src/testing/testing.go:561 +0x86d

goroutine 6 [runnable]:
gopkg.in/Clever/optimus.v3/sources/gearman.(*mockClient).Submit(0xc82000a740, 0x3e0ae0, 0x8, 0xc82000f500, 0x8, 0x8, 0x1160000, 0xc8200a6000, 0x1160030, 0xc8200aa000, ...)
        /Users/joshdeare/src/go/src/gopkg.in/Clever/optimus.v3/sources/gearman/gearman_test.go:25 +0xef
gopkg.in/Clever/optimus.v3/sources/gearman.(*table).start(0xc82000a7c0, 0x6b78b8, 0xc82000a740, 0x3e0ae0, 0x8, 0xc82000f500, 0x8, 0x8, 0xc82000a7a0)
        /Users/joshdeare/src/go/src/gopkg.in/Clever/optimus.v3/sources/gearman/gearman.go:57 +0x257
created by gopkg.in/Clever/optimus.v3/sources/gearman.New
        /Users/joshdeare/src/go/src/gopkg.in/Clever/optimus.v3/sources/gearman/gearman.go:76 +0x113
exit status 2
FAIL    gopkg.in/Clever/optimus.v3/sources/gearman      0.016s
make: *** [gopkg.in/Clever/optimus.v3/sources/gearman] Error 1
```

--- a/sources/gearman/gearman_test.go
+++ b/sources/gearman/gearman_test.go
@@ -2,14 +2,16 @@ package gearman
 
 import (
 	"fmt"
+	"io"
+	"testing"
+	"time"
+
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/mock"
 	"gopkg.in/Clever/gearman.v1/job"
 	"gopkg.in/Clever/gearman.v1/packet"
 	"gopkg.in/Clever/optimus.v3"
 	"gopkg.in/Clever/optimus.v3/tests"
-	"io"
-	"testing"
 )
 
 type mockClient struct {
@@ -54,6 +56,10 @@ func TestGearmanSource(t *testing.T) {
 		{"field1": "value2"},
 	}
 	go func() {
+		// Wait until a packet has been submitted
+		for len(c.chans) == 0 {
+			time.Sleep(time.Millisecond)
+		}
 		packets := c.chans[0]
 		packets <- handlePacket("", packet.WorkData, [][]byte{[]byte("1")})
 		packets <- handlePacket("", packet.WorkData, [][]byte{[]byte("2")})
@@ -71,6 +77,10 @@ func TestGearmanSourceFail(t *testing.T) {
 		return nil, nil
 	})
 	go func() {
+		// Wait until a packet has been submitted
+		for len(c.chans) == 0 {
+			time.Sleep(time.Millisecond)
+		}
 		packets := c.chans[0]
 		packets <- handlePacket("", packet.WorkWarning, [][]byte{[]byte("1")})
 		packets <- handlePacket("", packet.WorkFail, nil)

